apps/site/assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -62,16 +62,75 @@ const DeparturesAndMap = ({
     return false;
   };
 
+  const defaultPolylines = chain(routesWithPolylines)
+    .filter(
+      route =>
+        isASilverLineRoute(route) ||
+        isSubwayRoute(route) ||
+        isACommuterRailRoute(route)
+    )
+    .orderBy("sort_order", "desc")
+    .flatMap("polylines")
+    .uniqBy("id")
+    .value();
+
+  const DefaultRoutesMap = (): JSX.Element => {
+    return (
+      <StopMapRedesign stop={stop} lines={defaultPolylines} vehicles={[]} />
+    );
+  };
+
+  const SelectedRoutePatternMap = (): JSX.Element => {
+    const vehiclesForSelectedRoute = useVehiclesChannel(
+      departureInfo.departureRoute && departureInfo.departureDirectionId != null
+        ? {
+            routeId: departureInfo.departureRoute.id,
+            directionId: departureInfo.departureDirectionId
+          }
+        : null
+    );
+
+    const selectedRoute:
+      | RouteWithPolylines
+      | undefined = departureInfo.departureRoute
+      ? routesWithPolylines.find(
+          route => route.id === departureInfo.departureRoute!.id
+        )
+      : undefined;
```

Comment:
This might look nicer broken out into a helper function.  The signature of this variable is confusing.  Especially the
`| undefined = departureInfor.departureRoute` makes it looks like undefined is trying to be redefined.